src/client/Client.cc

```diff
@@ -8675,10 +8675,6 @@ int Client::_write(Fh *f, int64_t offset, uint64_t size, const char *buf,
   // use/adjust fd pos?
   if (offset < 0) {
     lock_fh_pos(f);
-    /*
-     * FIXME: this is racy in that we may block _after_ this point waiting for caps, and size may
-     * change out from under us.
-     */
```

Comment:
Well...we do fetch the size there rather early. What we probably could do is fetch extra caps if we know we'll need to do an O_APPEND write, and then delay setting up the offset until just before then, so that we don't lose them when we go to issue it.

That's a bigger patch though. I'll just drop this patch for now and just leave the comment there, but I'll also open a tracker bug to see if we can improve the O_APPEND semantics with a scheme like that.